### DISCUSSION TITLE
Custom payload

### DISCUSSION THREAD

**Discussion by Vircy** [rapid7/metasploit-framework#17153] [Q&A]
Hi i would like to use a payload variant obfuscated with msfvenom, can i use a .exe file as payload or i should use a different One ?

**Comment by zeroSteiner** [comment]
It depends entirely on how you want to deliver your payload and how you want to customize it. Some exploit modules take an EXE::Custom option, but modules that don't execute an EXE don't have that option but you can still specify your own shellcode using one of the generic payloads.
